services/dataAccessLayer.js

```diff
@@ -1,41 +1,62 @@
 const userQuery = require("../models/users");
 const members = require("../models/members");
 const { USER_SENSITIVE_DATA } = require("../constants/users");
+const AccessLevel = {
+  PUBLIC: "public",
+  INTERNAL: "internal",
+  PRIVATE: "private",
+  CONFIDENTIAL: "confidential",
+};
 
-const retrieveUsers = async ({ id = null, username = null, usernames = null, query = null, userdata }) => {
+const retrieveUsers = async ({
+  id = null,
+  username = null,
+  usernames = null,
+  query = null,
+  userdata,
+  level = AccessLevel.PUBLIC,
+  role = null,
+}) => {
   if (id || username) {
     let result;
     if (id != null) {
       result = await userQuery.fetchUser({ userId: id });
     } else {
       result = await userQuery.fetchUser({ username: username });
     }
-    removeSensitiveInfo(result.user);
+    const user = levelSpecificAccess(result.user, level, role);
+    result.user = user;
     return result;
   } else if (usernames) {
     const { users } = await userQuery.fetchUsers(usernames);
+    const result = [];
     users.forEach((element) => {
-      removeSensitiveInfo(element);
+      const user = levelSpecificAccess(element, level, role);
+      result.push(user);
     });
-    return users;
+    return result;
   } else if (query) {
     const { allUsers, nextId, prevId } = await userQuery.fetchPaginatedUsers(query);
+    const users = [];
     allUsers.forEach((element) => {
-      removeSensitiveInfo(element);
+      const user = levelSpecificAccess(element, level, role);
+      users.push(user);
     });
-    return { allUsers, nextId, prevId };
+    return { users, nextId, prevId };
   } else {
-    removeSensitiveInfo(userdata);
-    return userdata;
+    const result = await userQuery.fetchUser({ userId: userdata.id });
+    return levelSpecificAccess(result.user, level, role);
   }
 };
 
-const retrieveDiscordUsers = async () => {
+const retrieveDiscordUsers = async (level = AccessLevel.PUBLIC, role = null) => {
   const users = await userQuery.getDiscordUsers();
+  const userdata = [];
```

Comment:
this is storing all the user data can we call it in plural form `usersData` instead.